python/taichi/aot/utils.py

```diff
@@ -66,6 +66,8 @@ def produce_injected_args(kernel, symbolic_args=None):
                                   layout=Layout.AOS))
             else:
                 raise RuntimeError('')
+        elif isinstance(anno, MatrixType):
+            injected_args.append(anno)
```

Comment:
This is not quite right - when you compile you actually should feed in a dummy `Matrix` instance instead of `MatrixType` instance. (That's also why you need special handling for MatrixType args below...

I believe the right thing to do here is creating a Matrix of `anno` type and put each scalar in the inject_args one by one (once we have matrix type we can just inject one matrix arg but now it'll have to be  N scalars).